src/types/instance_definitions/bitwise_instance_def.rs

```diff
@@ -3,19 +3,19 @@ pub(crate) const INPUT_CELLS_PER_BITWISE: u32 = 2;
 
 #[derive(Clone, Debug, PartialEq)]
 pub(crate) struct BitwiseInstanceDef {
-    pub(crate) ratio: u32,
+    pub(crate) ratio: Option<u32>,
```

Comment:
I leave the choice to you because it's a compromise in ergonomics, but it may be a good idea to make this `Option<NonZeroU32>` instead. The same would be for the constructor.